platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStoreImpl.java

```diff
@@ -114,22 +114,16 @@ public Bytes getActiveRosterHash() {
         return latestRoundRosterPair.activeRosterHash();
     }
 
-    /** {@inheritDoc} */
     @Nullable
     @Override
-    public List<RoundRosterPair> getRosterHistory() {
-        final RosterState rosterStateSingleton = rosterState.get();
-        if (rosterStateSingleton != null
-                && !rosterStateSingleton.roundRosterPairs().isEmpty()) {
-            boolean allHashesPresent = rosterStateSingleton.roundRosterPairs().stream()
-                    .allMatch(pair -> rosterMap.get(ProtoBytes.newBuilder()
-                                    .value(pair.activeRosterHash())
-                                    .build())
-                            != null);
-            if (allHashesPresent) {
-                return rosterStateSingleton.roundRosterPairs();
-            }
-        }
-        return null;
+    public Bytes getPreviousRosterHash() {
+        final var rosterHistory = getRosterHistory();
+        return rosterHistory.size() > 1 ? rosterHistory.getLast().activeRosterHash() : null;
```

Comment:
This code makes a hard assumption that the size of the history is never greater than two. This logic will fail if the history is longer. And we do expect the size of the history to increase in the future when the full DAB is implemented.

Since we define the active roster as the entry at index 0, the previous roster should be defined as the entry at index 1 if it exists, or null otherwise. The logic in this method should never access entries with indices higher than 1.

Reply:
Ah got it, fixed.